packages/mwp-api-state/src/sync/index.js

```diff
@@ -74,35 +74,31 @@ export const getNavEpic = resolveRoutes => (action, store) => {
 	};
 	const cacheAction = requestMetadata.logout && { type: 'CACHE_CLEAR' };
 
-	const resolvePrevQueries = referrer.pathname
-		? resolveRoutes(referrer).then(getMatchedQueries(referrer))
-		: Promise.resolve([]);
-	const resolveNewQueries = resolveRoutes(location).then(
-		getMatchedQueries(location)
-	);
+	const previousQueries = referrer.pathname
+		? getMatchedQueries(referrer)(findMatches(referrer))
+		: [];
+	const newQueries = getMatchedQueries(location)(findMatches(location));
+	if (newQueries.filter(q => q).length === 0) {
+		// no valid queries - jump straight to 'complete'
+		return [api.complete([])];
+	}
+	// perform a fast comparison of previous route's serialized queries
+	// with the new route's serialized queries. All state refs for
+	// _shared_ queries should be retained
+	const serializedNew = newQueries.map(JSON.stringify);
+	const serializedPrev = previousQueries.map(JSON.stringify);
+	const sharedRefs = serializedPrev
+		.filter(qJSON => serializedNew.includes(qJSON))
+		.map(JSON.parse)
+		.map(q => q.ref);
+	requestMetadata.retainRefs = sharedRefs;
 
-	return Promise.all([resolveNewQueries, resolvePrevQueries]).then(
-		([newQueries, previousQueries]) => {
-			if (newQueries.filter(q => q).length === 0) {
-				// no valid queries - jump straight to 'complete'
-				return [api.complete([])];
-			}
-			// perform a fast comparison of previous route's serialized queries
-			// with the new route's serialized queries. All state refs for
-			// _shared_ queries should be retained
-			const serializedNew = newQueries.map(JSON.stringify);
-			const serializedPrev = previousQueries.map(JSON.stringify);
-			const sharedRefs = serializedPrev
-				.filter(qJSON => serializedNew.includes(qJSON))
-				.map(JSON.parse)
-				.map(q => q.ref);
-			requestMetadata.retainRefs = sharedRefs;
-			return [
-				cacheAction,
-				api.get(newQueries, requestMetadata),
-				clickActions.clear(),
-			].filter(a => a);
-		}
+	return Promise.resolve(
```

Comment:
I have some assumptions as to why Promise.all => Promise.resolve but would ♥️ hearing thoughts on the change.